komga-webui/src/functions/reader.ts

```diff
@@ -0,0 +1,396 @@
+import KomgaBooksService from '@/services/komga-books.service'
+import { BackgroundColors, ImageFit, ReadingDirection } from '@/types/enum-books'
+import { Map } from '@/functions/forms'
+import KomgaSeriesService from '@/services/komga-series.service'
+import { getBookTitleCompact } from '@/functions/book-title'
+import { Cookie, Mapper } from '@/functions/cookies'
+import { VueCookies } from 'vue-cookies'
+import { bookPageUrl } from '@/functions/urls'
+import { checkWebpFeature } from '@/functions/check-webp'
+import {
+  DoubleSpreadBuilder,
+  PageSpread,
+  PageSpreadBuilder, PageSpreads,
+  SingleSpreadBuilder,
+  WebToonBuilder,
+} from '@/functions/pages'
+import { VueRouter } from 'vue-router/types/router'
+
+export type Optional<V> = V | undefined
+
+export enum ReaderSettings {
+  DOUBLE_PAGES = 'webreader.doublePages',
+  SWIPE = 'webreader.swipe',
+  FIT = 'webreader.fit',
+  READ_DIR = 'webreader.readingDirection',
+  ANIMATIONS = 'webreader.animations',
+  BG_COLOR = 'webreader.background',
+}
+
+const TRUE_MAPPER = (v: any) => v === 'true'
+const IDENTITY_MAPPER = (v: any) => v
+
+async function tryCatch<V> (f: () => Promise<V>, defaultValue: V): Promise<V> {
+  return f().catch((r) => {
+    return defaultValue
+  })
+}
+
+export class Reader {
+  $komgaBooks: KomgaBooksService
+  $komgaSeries: KomgaSeriesService
+  $cookie: VueCookies
+  $router: VueRouter
```

Comment:
I have a concern with the class keeping a local copy of those services. It probably means the class is doing too much.